### DISCUSSION TITLE
[Homepage] Is there any way to use environment secrets?

### DISCUSSION THREAD

**Discussion by l-j** [tteck/Proxmox#2549] [Ideas]
I want to securely store credentials outside of my yaml configuration files. With Homepage & Docker, we can use the following mechanism:
https://gethomepage.dev/latest/installation/docker/#using-environment-secrets

As far as I can see, we cannot use it here at the moment.

## My solution
We can achieve a similar behaviour using systemd features, by adding to `[Service]` section of `/etc/systemd/system/homepage.service` the following line:
```
EnvironmentFile=-/opt/homepage/config/.env
```
and then the variables will be reloaded each time the homepage service is started/restarted when the `.env` file exists.

I am not sure if we want to add this change to the codebase, but at least someone will find this information useful.

**Comment by chelohomelab** [comment]
I tried doing this but not luck!
Tried with `EnvironmentFile=-/opt/homepage/config/.env` and with `EnvironmentFile=/opt/homepage/config/.env`, neither worked.
I have the file in place and the variables inside starts with HOMEPAGE_VAR_ as the docs recommends.
Anyone else having this issue?
